pymc_marketing/clv/models/gamma_gamma.py

```diff
@@ -56,7 +56,8 @@ def expected_customer_spend(
         mean_transaction_value, frequency = to_xarray(
             customer_id, mean_transaction_value, frequency
         )
-        assert self.idata is not None, "Model must be fitted first"
+        if self.idata is None:
+            raise ValueError("Model must be fitted first")
```

Comment:
This one is fine, but could be replaced by `posterior = self.fit_result`, then used below. Again that will take care of checking idata exists and raising a useful error.

The point of all this is to avoid code duplication of `model must be fit first`

Reply:
Solved in https://github.com/pymc-labs/pymc-marketing/pull/637/commits/89ca03516b41f0610e56f141fb9ce70f065b40a4